plugins/core/hu.bme.mit.gamma.expression.model/src/hu/bme/mit/gamma/expression/util/ExpressionModelValidator.java

```diff
@@ -637,6 +646,16 @@ else if (variableDeclarationType instanceof TypeReference &&
 										new ReferenceInfo(ExpressionModelPackage.Literals.INITIALIZABLE_ELEMENT__EXPRESSION, null)));
 							}
 						}
+						
+//////////////////////////////////////////////////////////////////////
+						// Array size must equal with number of array literal's elements
+						if (rhs.getOperands().size() != expressionEvaluator.evaluateInteger(arrayType.getSize())) {
```

Comment:
My concern still stands. If we try to evaluate a nonevaluable variable, then an exception is raised that is not handled by this method. The exception is not fatal though, we should handle (catch) it here as it indicates a case that cannot be handled (validated) at this level. (The user can use model checking if they are interested in such questions.)